app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    memo = {}
+
+    def wrapper(*args) -> Any:
+
+        if args in memo:
```

Comment:
Using `args` as a key in the `memo` dictionary assumes that all arguments are hashable. If any of the arguments are mutable (like lists or dictionaries), this will raise a `TypeError`. Consider ensuring that the arguments are hashable or converting them to a hashable type before using them as keys.